server/client_manager.py

```diff
@@ -290,23 +294,17 @@ def disemvowel_message(self, message):
     def __init__(self, server):
         self.clients = set()
         self.server = server
-        self.cur_id = [False] * self.server.config['playerlimit']
+        self.cur_id = [i for i in range(self.server.config['playerlimit'])]
         self.clients_list = []
 
     def new_client(self, transport):
-        cur_id = 0
-        for i in range(self.server.config['playerlimit']):
-                if not self.cur_id[i]:
-                    cur_id = i
-                    break
-        c = self.Client(self.server, transport, cur_id, self.server.get_ipid(transport.get_extra_info('peername')[0]))
+        c = self.Client(self.server, transport, heappop(self.cur_id), self.server.get_ipid(transport.get_extra_info('peername')[0]))
         self.clients.add(c)
-        self.cur_id[cur_id] = True
         return c
 
             
     def remove_client(self, client):
-        self.cur_id[client.id] = False
+        heappush(self.cur_id, client.id)
```

Comment:
Good thinking. It took me a while to understand what was going on here - basically get the smallest unused id in optimal time.